scripts/TRN(title matching).R

```diff
@@ -0,0 +1,135 @@
+## Script to compile all TRNs and titles of IV + EU trials to then sort by country/sponsor/etc for title matching
+# Generates pairings of EU and IV trials and their respective titles based on title matching algorithm
+
+##########################################################
+
+library(dplyr)
+library(tidyr)
+library(readr)
+library(here)
+library(fs)
+library(lubridate)
+library(stringr)
+library(stringdist)
+library(ctregistries)
+library(cli)
+
+dir_raw <- here("data", "raw")
+dir_processed <- here("data", "processed")
+
+# Download IV TRNs and titles. German full titles are here but the NCT titles here will need to be replaced by what's in studies.csv
+trials <- read_csv(path(dir_processed, "trials.csv"))
+
+# Download EU trial data dump, includes TRNs and full titles
+EU_dump <- read_csv(path(dir_raw, "euctr_euctr_dump-2024-02-03-054239.csv"))
+
+# Download studies.csv, which includes the 'official title' field for all NCTs
+studies <- read_csv(path(dir_raw,"studies.csv" ))
+
+##########################################################
+
+# extract IDs and Titles, and whatever information we want to filter by here for the EU trials.
+# Since IV trials don't have country info like EU trials, we will just get the ID and title here
+IV_ids <- trials |>
+          select(id,title) |>
+          unique()
+
+EU_ids <- EU_dump |>
+          select(eudract_number,
+                 member_state_concerned,
+                 national_competent_authority,
+                 full_title_of_the_trial)
```

Comment:
I selected both to add another layer of filtering before we title match. My logic was that getting both of these fields would ensure that we only looked at true German trials, and not any potential flukes. In the filtering step, both of these fields need to mention Germany to make it to the title-matching step. However, if they're always the same, it might make sense to only select one.